src/compiler/AST/Frontend.elm

```diff
@@ -85,41 +74,43 @@ recurse f expr =
             expr
 
         Plus e1 e2 ->
-            Plus (f e1) (f e2)
+            Plus
+                (Located.map f e1)
```

Comment:
We can probably do
```elm
recurse f expr =
  let
    f_ =
      Located.map f
  in
  ...
```
and then use that `f_`

Reply:
Also wait a minute ... Frontend `recurse` works on `Expr` but Typed `recurse` works on `LocatedExpr` :thinking:  Which one is right?

Reply:
it should be Frontend, it's only one Located.map from Parser that should be done in Frontend. Typed can't, because it has to deal with modules which carries Located

Reply:
yeah I've checked more in depth and looks like the types are alright after all :+1: it just raised a red flag in my head when I saw it first :)